### ISSUE TITLE
Sluggish crosshair after dynamic updates

### ISSUE THREAD

**Issue by carlos8f** [andredumas/techan.js#138]
Great library! Works well, except for sluggish performance after a while. Tried hardware acceleration on and off. Any advice would be appreciated.

https://zenbot.s8f.org/

**Comment by andredumas** [comment]
That's an interesting one. I observed the issue on that site, but when I attempt to inspect it becomes performant, even if i'm careful not to cause a redraw. Would be helpful if you could provide sample code of the isolated issue and any specific observations.

Nice color scheme btw

**Comment by carlos8f** [comment]
yeah, it's odd. Will see if I can reproduce the issue with minimal code. Thanks.

**Comment by carlos8f** [comment]
Page to reproduce the issue: http://bl.ocks.org/carlos8f/raw/ac8677ff6f6432b87b4d7fed25931e7d/

Minimal 1-line fix (comment out `selection.call(indicator)`: http://bl.ocks.org/carlos8f/raw/19670ee08bde4111a5606d44c42623a1/

CPU profile: https://gist.github.com/carlos8f/8a0d617739f35f7aa7a9af064c9fc76c#gistcomment-1852166

working on this at https://gist.github.com/carlos8f/8a0d617739f35f7aa7a9af064c9fc76c

**Comment by carlos8f** [comment]
There's a perf. bottleneck in the `scale()` function. https://gist.github.com/carlos8f/8a0d617739f35f7aa7a9af064c9fc76c#gistcomment-1852195

**Comment by carlos8f** [comment]
Sluggish issue is still present after patch, although my CPU profiler is no longer showing scale() as bottlenecks

**Comment by andredumas** [comment]
Nice pick up! Will give the pr a try see if it fixes for me.

Thank you.

**Comment by andredumas** [comment]
> Sluggish issue is still present after patch

Still a problem then?

**Comment by carlos8f** [comment]
still a problem after those date/string optimizations, just takes slightly longer to accumulate slowness.
